Boulder-Cryogenic-Quantum-Testbed/measurement
# Instrument and Temperature Control (#95)

* Add temperature control options to repository

Start by modifying the address for PNA-X
This handles the TCP address case
It is relevant for running from New York
Previous GPIB address works with Ontario

* Add temperature control class and logging

Uses PID control of the Janus gas handling system
Performs temperature sweeps and power sweeps
Next step is test the power sweeps
Currently testing single power measurements

* Working temperature sweep and PNA control

Running repeated measurements
Then doing a power sweep
Need to check things are OK later tonight

* Move PNA control into JanusTemperatureControl

Next step is try the power sweeps
The single powers appear to be working
Also, we have a better interrupt policy

* Add sparam input to control measurement type

Update temp_pid_ctrl file to handle many temps
Does not work -- will need to further debug
Also, the CMN MXC sensor does not work
Will need to manually reset this sensor

* Fix bug in zero current setting

Now the range and level set to zero
This prevents heating the MXC
And allows for easy temperature polling

* Fix bug in temperature controller and Ctrl-C exit

Add another entry to the out dict for tstamp
Next step is test the PNA-driver
And add powers, frequencies, electrical delays

* Fix PNA controller to toggle power

Next step is test the temperature control
Then combine temperature and VNA
Hopefully take data soon

* Add both VNA and temperature control together

Logs the temperature in the VNA filenames
Adds a thermalization step after settling
Also, takes data from the VNA

* Add temperatures, times, timestamps to logs

Add data to VNA folder names
Update thermalization time
Should be working now

* Remove multiprocessing threaded T-control

Add socket open / close for each temperature
This should helpfully address hang on read_cmn
Next step is to modify temperature logging
This should only write one file
The log will track all temperatures and times
Instead of creating a new file for each temperature

* Add socket initialization option, vna standalone

Fix bug in __del__()
Add power ranges up to 100 mA
Next step is to change the logging scheme

* Fix bug in pna_control.py to hold before reading

Add toggling power on / off between measurements
Add printing of powers and averages at each step
Next step is modify the temperature controller
This will have the appropriate logging in place
To store the temperatures at each time step
And perform temperature control while measuring

* Modify temperature logging, fix PNA hold

Previous fix did not wait for averaging
Now it waits for averaging + 3 seconds
This prevents premature / late writing

* Fix hang at end of power sweep with print

Add PID values to temperature logging
Add MXC heater current values to temperature log
Next step is to test the temperature controller
Look at three points separated in power

* Fix another bug in writing PID values from T-sweep

Add append function to write more data to log file
This allows an existing file to accumulate data
Add function to read the flow meter
Next step is to test the flow meter reader
And add the flow rate to the log file

* Add temperature reading of all stages, flow to log

Need to test the flow logging changes
Next step is try to control still heater remotely
Check the heater control channel settings

* Fix bugs in logging flow rate

Next step is add pressure reading
Also, log the other temperatures
And start outlining cooldown controller
To optimize the still heater current

* Fix thermalization timer and scale T_eps by Tset

Next steps are to log the pressures
And design a still heater controller
Based on the flow rate and the still temperature

* Add exception handling in TCP send / rcev

In the event of a network failure, do the following
Reset the Jacob control computer from the GHS
    1. Hold the OFF button on the UPS until it beeps
    2. Turn off the switch on the back of the GHS
    3. Wait about 1 minute
    4. Turn on the switch on the back of the GHS
    5. Hold the ON button on the UPS until it beeps
    6. Resume the cooldown circuit

* Add pressure reading to JanisCtrl class

Reads and prints all pressures by default
Returns pressure in V and mbar
Next step is to check flow rate and still
Change / control still heater current

* Reduce temperature precision on files, directories

This will ease the fitting of temperature data
So that all files and directories read xx mK
And avoids the need to correct for float rounding

* Add multiple resonator measurement to JanisCtrl

Measures at different frequencies, delays, spans
Similar to Corey Rae's multiple frequency user-file
Now integrated into the class object

* Updates before lab move
